Mathlib/Algebra/MonoidAlgebra/Defs.lean

```diff
@@ -77,38 +77,44 @@ def unexpander : Lean.PrettyPrinter.Unexpander
 end AddMonoidAlgebra
 
 namespace MonoidAlgebra
+
+@[inherit_doc MonoidAlgebra]
+scoped syntax:max (priority := high) term noWs "[" term "]" : term
+
+macro_rules | `($R[$M]) => `(MonoidAlgebra $R $M)
```

Comment:
How exactly does this work? What distinguishes whether `k[G]` gets parsed as `MonoidAlgebra k G` or `AddMonoidAlgebra k G`? Or are there some def-eq abuse shenanigans going on?